statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,538 @@
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, se_method, mse=False, replicates=None):
+        super().__init__(design, mse)
+        self.data = np.asarray(data)
+        if se_method == "jack":
+            self.est, self.vc = self.jack()
+            # self.vc = np.sqrt(np.diag(self.vc))
+        elif se_method == "boot":
+            self.est, self.vc = self.bootstrap(replicates)
+            self.vc = np.sqrt(self.vc)
+        else:
+            raise ValueError("Method %s not supported" % se_method)
+
+    def _stat(self, weights):
+        """
+        Returns calculation of mean.
+
+        Parameters
+        ----------
+        weights : np.array
+            The weights used to calculate the mean, will either be
+            original design weights or recalculated weights via jk,
+            boot, etc
+
+        Returns
+        -------
+        An array containing the statistic calculated on the columns
+        of the dataset.
+        """
+
+        # weights /= weights.sum()
+
+        return np.dot(weights, self.data) / np.sum(weights)
+
+
+class SurveyTotal(SurveyStat):
+    """
+    Calculates the total for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the total on
+    method: string
+        User inputs whether to get bootstrap or jackknife SE
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the mean on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vc : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, se_method,replicates=None, mse=False):
+        super().__init__(design, mse)
+        self.data = np.asarray(data)
+
+        if se_method == "jack":
+            self.est, self.vc = self.jack()
+            # self.vc = np.sqrt(np.diag(self.vc))
+        elif se_method == "boot":
+            self.est, self.vc = self.bootstrap(replicates)
+            self.vc = np.sqrt(self.vc)
+        else:
+            raise ValueError("Method %s not supported" % se_method)
+
+    def _stat(self, weights):
+        """
+        Returns calculation of mean.
+
+        Parameters
+        ----------
+        weights : np.array
+            The weights used to calculate the mean, will either be
+            original design weights or recalculated weights via jk,
+            boot, etc
+
+        Returns
+        -------
+        An array containing the statistic calculated on the columns
+        of the dataset.
+        """
+        return np.dot(weights, self.data)
+
+
+class SurveyQuantile(SurveyStat):
+    """
+    Calculates the quantiles[s] for each column.
```

Comment:
It's not clear to me that it's worth it to allow multiple quantiles in the same class instance.   It breaks the design compared to other statistics where there is only one statistic per instance.  There isn't much parallelism or recycled calculations so not clear that there is an efficiency advantage either.